addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/SitesTreeHelper.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.soap;
+
+import java.io.IOException;
+import javax.xml.soap.SOAPException;
+import javax.xml.soap.SOAPMessage;
+import org.parosproxy.paros.network.HttpMessage;
+import org.w3c.dom.NodeList;
+
+public class SitesTreeHelper {
+
+    public static String getNodeName(HttpMessage message) {
+        try {
+            SOAPMessage soapMsg =
+                    SoapMessageFactory.createMessage(message.getRequestBody().getBytes());
+            if (soapMsg == null) {
+                return null;
+            }
+            NodeList nodeList = soapMsg.getSOAPBody().getChildNodes();
+            StringBuilder leafName = new StringBuilder();
+            for (int i = 0; i < nodeList.getLength(); i++) {
+                if (nodeList.item(i).getLocalName() != null) {
+                    leafName.append(nodeList.item(i).getLocalName()).append(", ");
+                }
+            }
+            // Remove the extra ", " at the end.
+            leafName.setLength(leafName.length() - 2);
+            // Append SOAP Version.
+            leafName.append(
+                    message.getRequestHeader().getHeader("SOAPAction") == null
+                            ? " (v1.2)"
+                            : " (v1.1)");
+            return leafName.toString();
+        } catch (SOAPException | IOException e) {
+            return null;
+        }
```

Comment:
Should there be some sort of logged message here. Would it be better or safer to return empty string vs. null?

Reply:
Good question. We could return an empty string too, but in case that is not checked by the caller it could result in an empty site node. Still, it will certainly be safer to use an empty string. Is there a middle ground ;)?

Reply:
@psiinon @thc202 thoughts on this?

Reply:
If an empty name is not a valid name, then empty is better, but what matters most is that the behaviour is documented.

Re the exception being logged, I guess it could be logged as warn if that's not expected to happen frequently, otherwise debug.